lib/events/filelog.go

```diff
@@ -153,11 +158,51 @@ func (l *FileLog) EmitAuditEvent(ctx context.Context, event apievents.AuditEvent
 		return trace.NotFound(
 			"file log is not found due to permission or disk issue")
 	}
+
+	if len(line) > l.MaxScanTokenSize {
+		switch {
+		case canReduceMessageSize(event):
+			line, err = l.trimSizeAndMarshal(event)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+		default:
+			fields := log.Fields{"event_type": event.GetType(), "event_size": len(line)}
+			l.WithFields(fields).Warnf("Got a event that exeeded max allowed size.")
+			return trace.BadParameter("event size %q exceeds max entry size %q", len(line), l.MaxScanTokenSize)
+		}
+	}
+
 	// log it to the main log file:
 	_, err = fmt.Fprintln(l.file, string(line))
 	return trace.ConvertSystemError(err)
 }
 
+func canReduceMessageSize(event apievents.AuditEvent) bool {
+	_, ok := event.(messageSizeTrimmer)
+	return ok
+}
+
+func (l *FileLog) trimSizeAndMarshal(event apievents.AuditEvent) ([]byte, error) {
+	s, ok := event.(messageSizeTrimmer)
```

Comment:
I realize this is already merged, but there's no need to do the `messageSizeTrimmer` type assertion twice. It would be more efficient to call this `trimIfNecessary` or something, and just call it unconditionally.